rtengine/image16.h

```diff
@@ -40,62 +40,70 @@ class Image16 : public IImage16, public ImageIO
     Image16(int width, int height);
     ~Image16();
 
-    Image16*             copy();
+    Image16* copy() const;
 
-    Image8*              to8();
-    Imagefloat*          tofloat();
+    Image8* to8() const;
+    Imagefloat* tofloat() const;
 
-    virtual void         getStdImage(ColorTemp ctemp, int tran, Imagefloat* image, PreviewProps pp);
+    virtual void getStdImage(const ColorTemp &ctemp, int tran, Imagefloat* image, PreviewProps pp) const;
 
-    virtual const char*  getType() const
+    virtual const char* getType() const
     {
         return sImage16;
     }
-    virtual int          getBPS()
+    virtual int getBPS() const
     {
         return 8 * sizeof(unsigned short);
     }
-    virtual void         getScanline(int row, unsigned char* buffer, int bps, bool isFloat = false);
-    virtual void         setScanline(int row, unsigned char* buffer, int bps, unsigned int numSamples);
+    virtual void getScanline(int row, unsigned char* buffer, int bps, bool isFloat = false) const;
+    virtual void setScanline(int row, unsigned char* buffer, int bps, unsigned int numSamples);
 
     // functions inherited from IImage16:
-    virtual MyMutex&     getMutex()
+    virtual MyMutex& getMutex()
     {
         return mutex();
```

Comment:
Hmm, I don't get this one (at least from the diff): Are we returning a temporary here, or is that some non-intuitive `operator ()`? And why non-const?